services/redis.js

```diff
@@ -29,21 +29,36 @@ const connectionOptions = {
   }
 };
 
+const createClient = () => {
+  let client = redis.createClient(connectionOptions);
+
+  client.ping((err) => {
+    if (err) {
+      console.error('Can\'t ping the redis server!');
+
+      throw err;
+    }
+
+    debug('connection established');
+  });
+
+  return client;
+};
+
 module.exports = {
   connectionOptions,
-  createClient() {
-    let client = redis.createClient(connectionOptions);
-
-    client.ping((err) => {
-      if (err) {
-        console.error('Can\'t ping the redis server!');
+  createClient,
+  createClientFactory: () => {
```

Comment:
I saw you use a Factory Pattern to create client instances to redis. Wouldn’t we just want to have one client instance that connects to redis, and as such a singleton pattern would be more appropiate (or written in the factory pattern style but always returns the same instance)?